app/models/manageiq/providers/network_manager.rb

```diff
@@ -54,24 +45,12 @@ class << model_name
                :class_name  => "ManageIQ::Providers::BaseManager",
                :autosave    => true
 
-    has_many :availability_zones, -> { where.not(:ems_id => nil) }, :primary_key => :parent_ems_id, :foreign_key => :ems_id
-
-    # Relationships delegated to parent manager
-    delegate :cloud_tenants,
-             :flavors,
-             :cloud_resource_quotas,
-             :cloud_volumes,
+    # We cannot us a has many using :parent_ems_id primary key, since this doesn't belong to parent manager, so we need
+    # proper has_many :through or to delete these delegations.
```

Comment:
actually, cannot we just use has_many :through to avoid N+1?

That feels less fragile (well, you never know)